src/ResourceManager/Compute.ManagedService/Commands.Compute.ManagedService/StorageServices/NewAzureVhdVMCommand.cs

```diff
@@ -0,0 +1,576 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Compute.Common;
+using Microsoft.Azure.Commands.Compute.Models;
+using Microsoft.Azure.Commands.Compute.StorageServices;
+using Microsoft.Azure.Management.Compute;
+using Microsoft.Azure.Management.Compute.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using Microsoft.Azure.Management.Resources;
+using Microsoft.Azure.Management.Resources.Models;
+using Microsoft.Azure.Management.Storage;
+using Microsoft.Azure.Management.Storage.Models;
+using Microsoft.WindowsAzure.Commands.Sync.Download;
+using Microsoft.WindowsAzure.Commands.Tools.Vhd;
+using Microsoft.WindowsAzure.Commands.Tools.Vhd.Model;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+using Microsoft.WindowsAzure.Storage.Blob;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Compute
+{
+    [Cmdlet(VerbsCommon.New, ProfileNouns.VirtualHardDiskVirtualMachine, DefaultParameterSetName = DiskLinkParameterSetNameStr, SupportsShouldProcess = true)]
+    [OutputType(typeof(PSAzureOperationResponse))]
+    public class NewAzureVhdVMCommand : VirtualMachineBaseCmdlet
+    {
+        [Parameter(ParameterSetName = DiskLinkParameterSetNameStr, Mandatory = true, Position = 0, ValueFromPipelineByPropertyName = true)]
```

Comment:
This should follow the pattern of the creation cmdlet for New-AzureRmVM, the number of mandatory parameters should be minimized.  For New-AzureRmVM, I believe that only 'Name' and '(Admin)Credential' are required.  Thois should probably add the necessary disklink/diskfile parameters to this set.

Reply:
My suggestion for releasing this is that this belongs in a preview module for compute, and this is an additional parameter set for New-AzureRmVM